packages/agency/src/database/AgencyRepository.ts

```diff
@@ -1,44 +1,27 @@
-import { injectable } from 'inversify';
+import { inject, injectable } from 'inversify';
 import { TypeObjectId, IUser, IAgency } from '@donategifts/common';
 import { DBAgency } from './DBAgency';
 
 // TODO: needs typing
 @injectable()
 export class AgencyRepository {
+	constructor(@inject(DBAgency) private dbAgency: typeof DBAgency) {}
+
 	async getAgencyByUserId(userId: TypeObjectId<IUser>): Promise<IAgency | null> {
-		try {
-			return DBAgency.findOne({ accountManager: userId }).exec();
-		} catch (error) {
-			throw new Error(`Failed to get Agency: ${error}`);
-		}
+		return this.dbAgency.findOne({ accountManager: userId }).lean().exec();
 	}
 
 	async getAgencyWishCards(agencyId: TypeObjectId<IAgency>): Promise<IAgency | null> {
-		try {
-			return DBAgency.findOne({ _id: agencyId }).populate('wishCards').exec();
-		} catch (error) {
-			throw new Error(`Failed to get Wishcards: ${error}`);
-		}
+		return this.dbAgency.findOne({ _id: agencyId }).populate('wishCards').lean().exec();
```

Comment:
Ah, so we are going to check for it in Services, rather than using exceptions? :thinking: :+1:

Reply:
yes, i wanted to get rid of all the throws in the repositories since we catch everything usually in the service itself and the mongosetup right now logs exceptions anyways